app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt

```diff
@@ -129,10 +144,23 @@ class NewConversationViewModel
     private suspend fun searchPublic(searchTerm: String) {
         publicContactsSearchResult = ContactSearchResult.ExternalContact(SearchResultState.InProgress)
 
-        val result = withContext(Dispatchers.IO) {
+        val result: Result
+        val isFederationSearch = searchTerm.matches(FEDERATION_REGEX)
+        val query: String
+        val domain: String
+
+        if (isFederationSearch) {
```

Comment:
I think this is a good candidate to extract to seperate class and write unit tests for it